plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,566 @@
+        template = obj.get("template")
+        template_id = obj.get("template_id")
+
+        if scope_type == "bd":
+            schema_name = obj.get("schema")
+            schema_id = obj.get("schema_id")
+            if not schema_name and not schema_id:
+                mso.fail_json(msg="A member scope_type is bd and scope is used but the schema or schema_id option is missing.")
+            key = "schema-{0}-{1}-{2}".format(scope_type, schema_name, template if template else template_id)
+            mso_schema = obj_cache.get(key)
+            if not mso_schema:
+                mso_schema = MSOSchema(mso, schema_name, template, None, schema_id, template_id)
+                obj_cache[key] = mso_schema
+            mso_schema.set_template_bd(name, fail_module=True)
+            return mso_schema.schema_objects.get("template_bd").details.get("uuid")
+
+        if scope_type == "l3out":
+            key = "template-{0}-{1}".format(scope_type, template if template else template_id)
+            mso_template = obj_cache.get(key)
+            if not mso_template:
+                mso_template = MSOTemplate(mso, scope_type, template, template_id)
+                mso_template.validate_template(scope_type)
+                obj_cache[key] = mso_template
+            return mso_template.get_l3out_object(name=name, fail_module=True).details.get("uuid")
+
+    for member in members:
+        scope_type = member.get("scope_type")
+        track_member = {
+            "trackMember": {
+                "destIP": member.get("destination_ip"),
+                "scope": get_scope_obj_uuid(scope_type, member.get("scope_uuid"), member.get("scope")),
+                "scopeType": scope_type,
+                "ipslaMonitoringRef": mso_template.get_ipsla_monitoring_policy(
+                    uuid=member.get("ipsla_monitoring_policy_uuid"),
+                    name=member.get("ipsla_monitoring_policy").get("name") if member.get("ipsla_monitoring_policy") else None,
+                    fail_module=True,
+                ).details.get("uuid"),
+            },
+            "weight": member.get("weight"),
+        }
+        track_list_members.append(track_member)
+    return track_list_members
+
+
+def set_template_and_references(mso_template, ipsla_track_list_config):
+    l3out_scope_ref = {
+        "name": "scopeName",
+        "reference": "scope",
+        "template": "scopeTemplateName",
+        "templateId": "scopeTemplateId",
+        "type": "l3out",
+    }
+    bd_scope_ref = {
+        "name": "scopeName",
+        "reference": "scope",
+        "template": "scopeTemplateName",
+        "templateId": "scopeTemplateId",
+        "schema": "scopeSchemaName",
+        "schemaId": "scopeSchemaId",
+        "type": "bd",
+    }
+    ipsla_ref = {
+        "name": "ipslaMonitoringPolicyName",
+        "reference": "ipslaMonitoringRef",
+        "type": "ipslaMonitoringPolicy",
+    }
+    mso_template.clear_template_objects_cache()
```

Comment:
Are we using this here because the object related to uuid can be modified?

Reply:
There is the potential that the referenced object details could change between the initial match and after an update. However unlikely, this just makes sure the latest objects are used each time. The cache just helps when the same objects are referenced in multiple members.